dev/tools/localization/gen_l10n.dart

```diff
@@ -744,6 +491,19 @@ class LocalizationsGenerator {
     outputFile = _fs.file(path.join(l10nDirectory.path, outputFileString));
   }
 
+  static bool _isValidClassName(String className) {
+    // Dart class name cannot contain non-alphanumeric symbols
+    if (className.contains(RegExp(r'[^a-zA-Z_\d]')))
+      return false;
+    // Dart class name must start with upper case character
+    if (className[0].contains(RegExp(r'[a-z]')))
```

Comment:
Or underscore?

Reply:
In this case className must work as a public identifier so there shouldn't be a leading underscore. Added a check for that.